### PR TITLE
feat!: drop support for node16

### PR DESCRIPTION
**Pull request by Lightning00Blade**
A good opportunity to remove some deprecation if possible/old enough. 
I think we should cut a release before merging this.

### CODE REVIEW

File: packages/browsers/package.json [hunk 1]
```diff
@@ -89,7 +89,7 @@
   "author": "The Chromium Authors",
   "license": "Apache-2.0",
   "engines": {
-    "node": ">=16.3.0"
+    "node": ">=18.17.1"
```
Comment (jrandolf) [inline packages/browsers/package.json]: `>=18` everywhere.

**Review by jrandolf-google — CHANGES_REQUESTED** [review]

**Comment by jrandolf** [comment]
If you could, could you look back at v18 and remove anything that was deprecated?

**Comment by jrandolf** [comment]
Let's create an issue that outlines what we will be deprecating from Puppeteer. For breaking releases, we should document everything so people can also chime in.

**Comment by OrKoN** [comment]
I actually just remembered, we should double check the version of Node Chromium is on.

**Comment by OrKoN** [comment]
yeah https://source.chromium.org/chromium/chromium/src/+/main:third_party/node/update_node_binaries;l=18;drc=5964ce8eb626cc3c18f437f194a1e9234aebe8aa

**Comment by Lightning00Blade** [comment]
Note: This is required to update Rollup to v4

**Review by OrKoN — APPROVED** [review]

**Review by jrandolf-google — APPROVED** [review]

**Review by dandan3100 — APPROVED** [review]
GitBush/NObOdy.eth

**Pull request merged**